dds/DCPS/RTPS/ICE/EndpointManager.cpp

```diff
@@ -722,11 +722,35 @@ void EndpointManager::compute_active_foundations(ActiveFoundationSet& a_active_f
 
 void EndpointManager::check_invariants() const
 {
+  // Check for expired deferred checks.
+  size_t deferred = 0;
+  for (DeferredTriggeredChecksType::const_iterator pos = deferred_triggered_checks_.begin(),
+         limit = deferred_triggered_checks_.end(); pos != limit; ++pos) {
+    const DeferredTriggeredCheckListType& list = pos->second;
+
+    OPENDDS_ASSERT(!list.empty());
+    OPENDDS_ASSERT(list.front().expiration_date >= MonotonicTimePoint::now() - agent_impl->get_configuration().server_reflexive_address_period() - agent_impl->get_configuration().server_reflexive_address_period());
+    deferred += list.size();
+  }
+
   for (UsernameToChecklistType::const_iterator pos = username_to_checklist_.begin(),
-       limit = username_to_checklist_.end(); pos != limit; ++pos) {
+         limit = username_to_checklist_.end(); pos != limit; ++pos) {
     ChecklistPtr checklist = pos->second;
+    OPENDDS_ASSERT(checklist->original_remote_agent_info().username == pos->first);
     checklist->check_invariants();
   }
+
+  for (TransactionIdToChecklistType::const_iterator pos = transaction_id_to_checklist_.begin(),
+         limit = transaction_id_to_checklist_.end(); pos != limit; ++pos) {
+    ChecklistPtr checklist = pos->second;
```

Comment:
Is this going to be an "unused variable" warning when asserts are disabled?  (same on 751)